### PR TITLE
Union branchs

### PR DESCRIPTION
**Pull request by JosueRodriguezGarcia**

### CODE REVIEW

File: config.properties [hunk 1]
```diff
@@ -0,0 +1,5 @@
+## Trello Autentication EndPoint
+consumerKey = 0c5256a20e5f26b03836fa01c1ac68c3
+consumerSecret = 94f824bca09faa7d28eec919171da5e89de0552dceae1e7e03db239ba190ed67
+accessToken = e3228d269f332facfbc1a638a1874bfb6d6a09173469968bcb419e165d926142
+tokenSecret = e3228d269f332facfbc1a638a1874bfb6d6a09173469968bcb419e165d926142
```
Comment (aburgosu) [inline config.properties]: Insert final white line.

File: src/test/java/com/foundation/trello/ReadConfigurationTest.java [hunk 2]
```diff
@@ -0,0 +1,146 @@
...
+     */
+    @Test
+    public void readConfigurationFileTokenSecret() {
+        String actual = tokenSecret;
+        String expected = "e3228d269f332facfbc1a638a1874bfb6d6a09173469968bcb419e165d926142";
+        Assert.assertEquals(actual, expected);
+    }
+
+    /**
+     * this test verify that the GET return a state 200.
```
Comment (aburgosu) [inline src/test/java/com/foundation/trello/ReadConfigurationTest.java]: Start the sentence with Capital letter.

File: src/test/java/com/foundation/trello/ReadConfigurationTest.java [hunk 3]
```diff
@@ -0,0 +1,146 @@
...
+                auth().
+                oauth(consumerKey, consumerSecret, accessToken, tokenSecret).
+                when().
+                post("https://api.trello.com/1/boards/").
+                then().assertThat().statusCode(200).log().all().
+                extract().response();
+    }
+
+    /**
+     * this test verify that the DELETE return a state 200.
```
Comment (aburgosu) [inline src/test/java/com/foundation/trello/ReadConfigurationTest.java]: Start the sentence with capital letter.

File: src/test/java/com/foundation/trello/ReadConfigurationTest.java [hunk 4]
```diff
@@ -0,0 +1,146 @@
...
+                auth().
+                oauth(consumerKey, consumerSecret, accessToken, tokenSecret).
+                when().
+                delete("https://api.trello.com/1/boards/AdPLEOdq").
+                then().assertThat().statusCode(200).
+                extract().response();
+    }
+
+    /**
+     * this test verify that the POST return a state 200.
```
Comment (aburgosu) [inline src/test/java/com/foundation/trello/ReadConfigurationTest.java]: Start the sentence with capital letter.

File: config.properties (hunk 1 above)
Comment (JosueRodriguezGarcia) [reply config.properties]: Fixed, it's already added to the line.

File: src/test/java/com/foundation/trello/ReadConfigurationTest.java (hunk 2 above)
Comment (JosueRodriguezGarcia) [reply src/test/java/com/foundation/trello/ReadConfigurationTest.java]: fixed

File: src/test/java/com/foundation/trello/ReadConfigurationTest.java (hunk 3 above)
Comment (JosueRodriguezGarcia) [reply src/test/java/com/foundation/trello/ReadConfigurationTest.java]: Fixed.

File: src/test/java/com/foundation/trello/ReadConfigurationTest.java (hunk 4 above)
Comment (JosueRodriguezGarcia) [reply src/test/java/com/foundation/trello/ReadConfigurationTest.java]: Fixed.

File: src/main/java/com/foundation/trello/ReadConfiguration.java [hunk 5]
```diff
@@ -0,0 +1,34 @@
+package com.foundation.trello;
+
+import java.io.FileInputStream;
+import java.io.IOException;
+import java.io.InputStream;
+import java.util.Properties;
+
+/**
+ * This class read the config.properties file.
```
Comment (MadayAlcala) [inline src/main/java/com/foundation/trello/ReadConfiguration.java]: Please change the verb to third person 'reads'.

File: src/test/java/com/foundation/trello/ReadConfigurationTest.java [hunk 6]
```diff
@@ -0,0 +1,146 @@
...
+    /**
+     * This method setup the uri.
+     */
+    @Before
+    public void setUp() {
+
+    }
+
+    /**
+     * This test verify that the consumerKey is correct get of the properties file.
```
Comment (MadayAlcala) [inline src/test/java/com/foundation/trello/ReadConfigurationTest.java]: Change the verb to third person, it should be --> 'verifies'.

File: src/test/java/com/foundation/trello/ReadConfigurationTest.java [hunk 7]
```diff
@@ -0,0 +1,146 @@
...
+     */
+    @Test
+    public void readConfigurationFileConsumerKey() {
+        String actual = consumerKey;
+        String expected = "0c5256a20e5f26b03836fa01c1ac68c3";
+        Assert.assertEquals(actual, expected);
+    }
+
+    /**
+     * This test verify that the consumerSecret is correct get of the properties file.
```
Comment (MadayAlcala) [inline src/test/java/com/foundation/trello/ReadConfigurationTest.java]: Change the verb to third person, it should be --> 'verifies'.

File: src/test/java/com/foundation/trello/ReadConfigurationTest.java [hunk 8]
```diff
@@ -0,0 +1,146 @@
...
+     */
+    @Test
+    public void readConfigurationFileConsumerSecret() {
+        String actual = consumerSecret;
+        String expected = "94f824bca09faa7d28eec919171da5e89de0552dceae1e7e03db239ba190ed67";
+        Assert.assertEquals(actual, expected);
+    }
+
+    /**
+     * This test verify that the accessToken is correct get of the properties file.
```
Comment (MadayAlcala) [inline src/test/java/com/foundation/trello/ReadConfigurationTest.java]: Change the verb to third person, it should be --> 'verifies'.

File: src/test/java/com/foundation/trello/ReadConfigurationTest.java [hunk 9]
```diff
@@ -0,0 +1,146 @@
...
+     */
+    @Test
+    public void readConfigurationFileAccessToken() {
+        String actual = accessToken;
+        String expected = "e3228d269f332facfbc1a638a1874bfb6d6a09173469968bcb419e165d926142";
+        Assert.assertEquals(actual, expected);
+    }
+
+    /**
+     * This test verify that the tokenSecret is correct get of the properties file.
```
Comment (MadayAlcala) [inline src/test/java/com/foundation/trello/ReadConfigurationTest.java]: Change the verb to third person, it should be --> 'verifies'.

File: src/test/java/com/foundation/trello/ReadConfigurationTest.java (hunk 2 above)
Comment (MadayAlcala) [reply src/test/java/com/foundation/trello/ReadConfigurationTest.java]: Please change the verb to third person, it should be --> 'verifies'.

File: src/test/java/com/foundation/trello/ReadConfigurationTest.java [hunk 10]
```diff
@@ -0,0 +1,146 @@
...
+     */
+    @Test
+    public void readConfigurationFileTokenSecret() {
+        String actual = tokenSecret;
+        String expected = "e3228d269f332facfbc1a638a1874bfb6d6a09173469968bcb419e165d926142";
+        Assert.assertEquals(actual, expected);
+    }
+
+    /**
+     * This test verify that the GET return a state 200.
```
Comment (MadayAlcala) [inline src/test/java/com/foundation/trello/ReadConfigurationTest.java]: Change the verb to third person, it should be --> 'verifies'.

File: src/test/java/com/foundation/trello/ReadConfigurationTest.java (hunk 6 above)
Comment (JosueRodriguezGarcia) [reply src/test/java/com/foundation/trello/ReadConfigurationTest.java]: Okay, it's been modified.

File: src/test/java/com/foundation/trello/ReadConfigurationTest.java [hunk 11]
```diff
@@ -0,0 +1,146 @@
...
+                auth().
+                oauth(consumerKey, consumerSecret, accessToken, tokenSecret).
+                        when().
+                get("https://api.trello.com/1/boards/9reOdft6").
+                then().assertThat().statusCode(200).log().all().
+                extract().response();
+    }
+
+    /**
+     * This test verify that the PUT return a state 200.
```
Comment (MadayAlcala) [inline src/test/java/com/foundation/trello/ReadConfigurationTest.java]: Change the verb to third person, it should be --> 'verifies'.

File: src/test/java/com/foundation/trello/ReadConfigurationTest.java (hunk 7 above)
Comment (JosueRodriguezGarcia) [reply src/test/java/com/foundation/trello/ReadConfigurationTest.java]: Okay, it's been modified.

File: src/test/java/com/foundation/trello/ReadConfigurationTest.java [hunk 12]
```diff
@@ -0,0 +1,146 @@
...
+                oauth(consumerKey, consumerSecret, accessToken, tokenSecret).
+                param("name", "HelloWorld").log().all().
+                when().
+                put("https://api.trello.com/1/boards/9reOdft6").
+                then().assertThat().statusCode(200).log().all().
+                extract().response();
+    }
+
+    /**
+     * This test verify that the POST return a state 200.
```
Comment (MadayAlcala) [inline src/test/java/com/foundation/trello/ReadConfigurationTest.java]: Change the verb to third person, it should be --> 'verifies'.

File: src/test/java/com/foundation/trello/ReadConfigurationTest.java (hunk 8 above)
Comment (JosueRodriguezGarcia) [reply src/test/java/com/foundation/trello/ReadConfigurationTest.java]: Okay, it's been modified.

File: src/test/java/com/foundation/trello/ReadConfigurationTest.java (hunk 9 above)
Comment (JosueRodriguezGarcia) [reply src/test/java/com/foundation/trello/ReadConfigurationTest.java]: Okay, it's been modified.

File: src/test/java/com/foundation/trello/ReadConfigurationTest.java (hunk 10 above)
Comment (JosueRodriguezGarcia) [reply src/test/java/com/foundation/trello/ReadConfigurationTest.java]: Okay, it's been modified.

File: src/test/java/com/foundation/trello/ReadConfigurationTest.java (hunk 11 above)
Comment (JosueRodriguezGarcia) [reply src/test/java/com/foundation/trello/ReadConfigurationTest.java]: Okay, it's been modified.

File: src/test/java/com/foundation/trello/ReadConfigurationTest.java [hunk 13]
```diff
@@ -0,0 +1,146 @@
...
+                auth().
+                oauth(consumerKey, consumerSecret, accessToken, tokenSecret).
+                when().
+                delete("https://api.trello.com/1/boards/AdPLEOdq").
+                then().assertThat().statusCode(200).
+                extract().response();
+    }
+
+    /**
+     * this test verify that the POST return a state 200.
```
Comment (MadayAlcala) [inline src/test/java/com/foundation/trello/ReadConfigurationTest.java]: Change the verb to third person, it should be --> 'verifies'.

File: src/test/java/com/foundation/trello/ReadConfigurationTest.java (hunk 12 above)
Comment (JosueRodriguezGarcia) [reply src/test/java/com/foundation/trello/ReadConfigurationTest.java]: Okay, it's been modified.

File: src/test/java/com/foundation/trello/ReadConfigurationTest.java [hunk 14]
```diff
@@ -63,86 +70,76 @@ public void readConfigurationFileTokenSecret() {
     }
 
     /**
-     *  this test verify that the GET return a state 200.
+     * this test verify that the GET return a state 200.
```
Comment (LimbertVargas) [inline src/test/java/com/foundation/trello/ReadConfigurationTest.java]: Uppercase in the first letter, please.

File: src/test/java/com/foundation/trello/ReadConfigurationTest.java [hunk 15]
```diff
@@ -63,86 +70,76 @@ public void readConfigurationFileTokenSecret() {
...
                         when().
-                get("/1/boards/9reOdft6").
+                get("https://api.trello.com/1/boards/9reOdft6").
                 then().assertThat().statusCode(200).log().all().
                 extract().response();
     }
 
     /**
-     *  this test verify that the PUT return a state 200.
+     * this test verify that the PUT return a state 200.
```
Comment (LimbertVargas) [inline src/test/java/com/foundation/trello/ReadConfigurationTest.java]: Uppercase in the first letter, please.

File: src/test/java/com/foundation/trello/ReadConfigurationTest.java [hunk 16]
```diff
@@ -63,86 +70,76 @@ public void readConfigurationFileTokenSecret() {
...
                 when().
-                post("/1/boards/").
+                post("https://api.trello.com/1/boards/").
                 then().assertThat().statusCode(200).log().all().
                 extract().response();
     }
 
     /**
-     *  this test verify that the DELETE return a state 200.
+     * this test verify that the DELETE return a state 200.
```
Comment (LimbertVargas) [inline src/test/java/com/foundation/trello/ReadConfigurationTest.java]: Uppercase in the first letter, please.

File: src/test/java/com/foundation/trello/ReadConfigurationTest.java [hunk 17]
```diff
@@ -63,86 +70,76 @@ public void readConfigurationFileTokenSecret() {
...
                 when().
-                delete("/1/boards/AdPLEOdq").
+                delete("https://api.trello.com/1/boards/AdPLEOdq").
                 then().assertThat().statusCode(200).
                 extract().response();
     }
 
     /**
-     *  this test verify that the POST return a state 200.
+     * this test verify that the POST return a state 200.
```
Comment (LimbertVargas) [inline src/test/java/com/foundation/trello/ReadConfigurationTest.java]: Uppercase in the first letter, please.

File: src/test/java/com/foundation/trello/ReadConfigurationTest.java [hunk 18]
```diff
@@ -0,0 +1,146 @@
...
+    private ReadConfiguration reader = new ReadConfiguration();
+    private String consumerKey = reader.readConfigurationFile("consumerKey");
+    private String consumerSecret = reader.readConfigurationFile("consumerSecret");
+    private String accessToken = reader.readConfigurationFile("accessToken");
+    private String tokenSecret = reader.readConfigurationFile("tokenSecret");
+
+    /**
+     * This method setup the uri.
+     */
+    @Before
```
Comment (MadayAlcala) [inline src/test/java/com/foundation/trello/ReadConfigurationTest.java]: Please remove this unusable method.

File: src/test/java/com/foundation/trello/ReadConfigurationTest.java (hunk 14 above)
Comment (JosueRodriguezGarcia) [reply src/test/java/com/foundation/trello/ReadConfigurationTest.java]: Okay, it's been modified.

File: src/test/java/com/foundation/trello/ReadConfigurationTest.java (hunk 15 above)
Comment (JosueRodriguezGarcia) [reply src/test/java/com/foundation/trello/ReadConfigurationTest.java]: Okay, it's been modified.

File: src/test/java/com/foundation/trello/ReadConfigurationTest.java (hunk 16 above)
Comment (JosueRodriguezGarcia) [reply src/test/java/com/foundation/trello/ReadConfigurationTest.java]: Okay, it's been modified.

File: src/test/java/com/foundation/trello/ReadConfigurationTest.java (hunk 17 above)
Comment (JosueRodriguezGarcia) [reply src/test/java/com/foundation/trello/ReadConfigurationTest.java]: Okay, it's been modified.

File: src/test/java/com/foundation/trello/ReadConfigurationTest.java (hunk 18 above)
Comment (JosueRodriguezGarcia) [reply src/test/java/com/foundation/trello/ReadConfigurationTest.java]: ok, fixed

File: src/test/java/com/foundation/trello/ReadConfigurationTest.java [hunk 19]
```diff
@@ -0,0 +1,138 @@
...
+import cucumber.api.java.Before;
+import io.restassured.http.ContentType;
+import org.testng.Assert;
+import org.testng.annotations.Test;
+
+import java.util.HashMap;
+import java.util.Map;
+
+import static io.restassured.RestAssured.given;
+
```
Comment (MadayAlcala) [inline src/test/java/com/foundation/trello/ReadConfigurationTest.java:11]: Please delete unnecessary line.

File: src/main/java/com/foundation/trello/ReadConfiguration.java (hunk 5 above)
Comment (MadayAlcala) [inline src/main/java/com/foundation/trello/ReadConfiguration.java]: Please use third person for 'read'.

File: src/test/java/com/foundation/trello/ReadConfigurationTest.java (hunk 19 above)
Comment (JosueRodriguezGarcia) [reply src/test/java/com/foundation/trello/ReadConfigurationTest.java:11]: fixed

File: src/main/java/com/foundation/trello/ReadConfiguration.java (hunk 5 above)
Comment (JosueRodriguezGarcia) [reply src/main/java/com/foundation/trello/ReadConfiguration.java]: fixed

File: src/test/java/com/foundation/trello/ReadConfigurationTest.java (hunk 13 above)
Comment (JosueRodriguezGarcia) [reply src/test/java/com/foundation/trello/ReadConfigurationTest.java]: fixed

**Review by MadayAlcala — APPROVED** [review]
Ok, it is ready to merge.

**Review by aburgosu — APPROVED** [review]

**Pull request merged**